### PR TITLE
Fixes #5055: infinite loop in lcdDrawFilledRect() when `h` parameter …

### LINKED ISSUE #5055: Lua script causes q x7 to reboot
I was attempting to use this script for telemetry purposes, and found that it crashes and reboots radio upon execution.

I believe it is related to the draw commands, but I am not certain. Removing several commands and restructuring the script made it work on the x7, but I wanted to report it.

[dlvector-v1_1.lua.txt](https://github.com/opentx/opentx/files/1098838/dlvector-v1_1.lua.txt)

https://www.rcgroups.com/forums/showpost.php?p=34749134&postcount=49435

### PR DESCRIPTION
**Pull request by projectkk2glider**
…is bigger than 127. Similar fixes in other places.

Closes #5055

### CODE REVIEW

**Comment by bsongis** [comment]
I will review it, I just want to check carefully how much flash it takes on AVR 9X

**Comment by projectkk2glider** [comment]
Rebased because of the conflicts.

**Comment by 3djc** [comment]
@bsongis:
**Current next:**

Program:   52058 bytes (79.4% Full)
(.text + .data + .bootloader)
Data:       2887 bytes (70.5% Full)
(.data + .bss + .noinit)

**This PR**

Device: atmega64
Program:   52022 bytes (79.4% Full)
(.text + .data + .bootloader)
Data:       2887 bytes (70.5% Full)
(.data + .bss + .noinit)

File: radio/src/gui/128x64/lcd.cpp [hunk 1]
```diff
@@ -750,12 +750,12 @@ void lcdDrawRect(coord_t x, coord_t y, coord_t w, coord_t h, uint8_t pat, LcdFla
 void lcdDrawFilledRect(coord_t x, scoord_t y, coord_t w, coord_t h, uint8_t pat, LcdFlags att)
 {
 #if defined(CPUM64)
-  for (scoord_t i=y; i<y+h; i++) {
+  for (scoord_t i=y; i<(scoord_t)(y+h); i++) {
```
Comment (bsongis) [inline radio/src/gui/128x64/lcd.cpp:753]: cosmetics: `scoord_t(y+h)`

**Review by bsongis — APPROVED** [review]

**Comment by bsongis** [comment]
I was wrong in saying it would use more flash, we only call the invert line function with a constant that the compiler knows and it will optimize out the check.

**Pull request merged**